pkg/pipeline/ingest/ingest_kafka_test.go

```diff
@@ -0,0 +1,212 @@
+
+	config.Opt.PipeLine.Ingest.Kafka = string(b)
+	config.Opt.PipeLine.Ingest.Type = "kafka"
+	newIngest, err := NewIngestKafka()
+	require.Equal(t, err, nil)
+	return newIngest
+}
+
+func Test_NewIngestKafka1(t *testing.T) {
+	newIngest := initNewIngestKafka(t, testConfig1)
+	ingestKafka := newIngest.(*ingestKafka)
+	require.Equal(t, "topic1", ingestKafka.kafkaParams.Topic)
+	require.Equal(t, "group1", ingestKafka.kafkaParams.GroupId)
+	expectedBrokers := []string{"1.1.1.1:9092"}
+	require.Equal(t, expectedBrokers, ingestKafka.kafkaParams.Brokers)
+	require.Equal(t, "FirstOffset", ingestKafka.kafkaParams.StartOffset)
+	require.Equal(t, 2, len(ingestKafka.kafkaReader.Config().GroupBalancers))
+	require.Equal(t, int64(300), ingestKafka.kafkaParams.BatchReadTimeout)
+}
+
+func Test_NewIngestKafka2(t *testing.T) {
+	newIngest := initNewIngestKafka(t, testConfig2)
+	ingestKafka := newIngest.(*ingestKafka)
+	require.Equal(t, "topic2", ingestKafka.kafkaParams.Topic)
+	require.Equal(t, "group2", ingestKafka.kafkaParams.GroupId)
+	expectedBrokers := []string{"1.1.1.2:9092"}
+	require.Equal(t, expectedBrokers, ingestKafka.kafkaParams.Brokers)
+	require.Equal(t, "LastOffset", ingestKafka.kafkaParams.StartOffset)
+	require.Equal(t, 1, len(ingestKafka.kafkaReader.Config().GroupBalancers))
+	require.Equal(t, defaultBatchReadTimeout, ingestKafka.kafkaParams.BatchReadTimeout)
+}
+
+var receivedEntries []interface{}
+var dummyChan chan bool
+
+func dummyProcessFunction(entries []interface{}) {
+	receivedEntries = entries
+	dummyChan <- true
+}
+
+func Test_IngestKafka(t *testing.T) {
+	dummyChan = make(chan bool)
+	newIngest := initNewIngestKafka(t, testConfig1)
+	ingestKafka := newIngest.(*ingestKafka)
+
+	// run Ingest in a separate thread
+	go func() {
+		ingestKafka.Ingest(dummyProcessFunction)
+	}()
+	// wait a second for the ingest pipeline to come up
+	time.Sleep(time.Second)
+
+	// feed some data into the pipeline
+	record1 := "{\"Bytes\":20801,\"DstAddr\":\"10.130.2.1\",\"DstPort\":36936,\"Packets\":401,\"SrcAddr\":\"10.130.2.13\",\"SrcPort\":3100}"
+	record2 := "{\"Bytes\":20802,\"DstAddr\":\"10.130.2.2\",\"DstPort\":36936,\"Packets\":402,\"SrcAddr\":\"10.130.2.13\",\"SrcPort\":3100}"
+	record3 := "{\"Bytes\":20803,\"DstAddr\":\"10.130.2.3\",\"DstPort\":36936,\"Packets\":403,\"SrcAddr\":\"10.130.2.13\",\"SrcPort\":3100}"
+
+	inChan := ingestKafka.in
+	inChan <- record1
+	inChan <- record2
+	inChan <- record3
+
+	// wait for the data to have been processed
+	<-dummyChan
+
+	require.Equal(t, 3, len(receivedEntries))
+	require.Equal(t, record1, receivedEntries[0])
+	require.Equal(t, record2, receivedEntries[1])
+	require.Equal(t, record3, receivedEntries[2])
+
+	// make the ingest thread exit
+	ingestKafka.exitChan <- true
+	time.Sleep(time.Second)
+}
+
+type fakeKafkaReader struct {
+	mock.Mock
+}
+
+var fakeRecord = []byte(`{"Bytes":20801,"DstAddr":"10.130.2.1","DstPort":36936,"Packets":401,"SrcAddr":"10.130.2.13","SrcPort":3100}`)
```

Comment:
@KalmanMeth I miss a test that is able to show how we are moving from this []byte message in kafka into a json message ... this is part of the decode phase ... so maybe you can add this test in "pipeline_test". 

I want to make sure that we can handle this string --> json all the way from kafaka so that later we can continue with the regular pipeline phases of transformation etc ....